File: ctu_marketplace_new-feature-introduction_footer/ctu-marketplace-v2/src/main/java/com/ctu/marketplace/service/ResearchGroupService.java
package com.ctu.marketplace.service;

import java.util.List;

import com.ctu.marketplace.dto.request.ResearchGroupDto;
import com.ctu.marketplace.entity.ResearchGroup;

public interface ResearchGroupService {
    public ResearchGroup getById(Long researchGroupId);

    public List<ResearchGroup> getAll();

    public List<ResearchGroup> getAllByUserAdmin(Long userAdminId);

    public ResearchGroup create(ResearchGroupDto researchGroupDto);

    public ResearchGroup update(Long researchGroupId, ResearchGroupDto researchGroupDto);

    public boolean deleteById(Long researchGroupId);
}

File: ctu_marketplace_new-feature-introduction_footer/ctu-marketplace-web/src/services/researchGroup.service.js
import api from '../api/researchGroup';
import { RESEARCH_GROUP_URL } from '././../environments/constraints';

class ResearchGroupService {
  
    getAll() {
        return api.get(RESEARCH_GROUP_URL);
    }
    
    getById(id) {
        return api.get(`${RESEARCH_GROUP_URL}/${id}`);
    }

    create(data) {
        return api.post(RESEARCH_GROUP_URL, data);
    }

    update(id, data) {
        return api.put(`${RESEARCH_GROUP_URL}/${id}`, data);
    }

    delete(id) {
        return api.delete(`${RESEARCH_GROUP_URL}/${id}`);
    }

}

export default new ResearchGroupService();
